Mage.Sets/src/mage/cards/f/Fiendlash.java

```diff
@@ -0,0 +1,137 @@
+package mage.cards.f;
+
+import java.util.UUID;
+
+import mage.abilities.Ability;
+import mage.abilities.TriggeredAbilityImpl;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.costs.mana.ManaCostsImpl;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.continuous.BoostEquippedEffect;
+import mage.abilities.effects.common.continuous.GainAbilityAttachedEffect;
+import mage.abilities.keyword.EquipAbility;
+import mage.abilities.keyword.ReachAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.AttachmentType;
+import mage.constants.CardType;
+import mage.constants.Outcome;
+import mage.constants.SubType;
+import mage.constants.Zone;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+import mage.players.Player;
+import mage.target.common.TargetPlayerOrPlaneswalker;
+
+/**
+ *
+ * @author zeffirojoe
+ */
+public final class Fiendlash extends CardImpl {
+
+    public Fiendlash(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[] { CardType.ARTIFACT }, "{1}{R}");
+
+        this.subtype.add(SubType.EQUIPMENT);
+
+        // Equipped creature gets +2/+0 and has reach.
+        Ability staticAbility = new SimpleStaticAbility(new BoostEquippedEffect(2, 0));
+        staticAbility.addEffect(new GainAbilityAttachedEffect(ReachAbility.getInstance(), AttachmentType.EQUIPMENT)
+                .setText("and has reach"));
+        this.addAbility(staticAbility);
+
+        // Whenever equipped creature is dealt damage, it deals damage equal to its
+        // power to target player or planeswalker.
+        this.addAbility(new FiendlashTriggeredAbility());
+
+        // Equip {2}{R}
+        this.addAbility(new EquipAbility(Outcome.AddAbility, new ManaCostsImpl<>("{2}{R}")));
+    }
+
+    private Fiendlash(final Fiendlash card) {
+        super(card);
+    }
+
+    @Override
+    public Fiendlash copy() {
+        return new Fiendlash(this);
+    }
+}
+
+class FiendlashTriggeredAbility extends TriggeredAbilityImpl {
+
+    FiendlashTriggeredAbility() {
+        super(Zone.BATTLEFIELD, new FiendlashEffect(), false);
+        this.addTarget(new TargetPlayerOrPlaneswalker());
+    }
+
+    private FiendlashTriggeredAbility(final FiendlashTriggeredAbility ability) {
+        super(ability);
+    }
+
+    @Override
+    public FiendlashTriggeredAbility copy() {
+        return new FiendlashTriggeredAbility(this);
+    }
+
+    @Override
+    public boolean checkEventType(GameEvent event, Game game) {
+        return event.getType() == GameEvent.EventType.DAMAGED_PERMANENT;
```

Comment:
If you look at `DealtDamageToSourceTriggeredAbility`, it uses `COMBAT_DAMAGE_STEP_POST` so that the trigger is only fired once per combat (it shouldn't trigger multiple times if more than one blocker is assigned to it.)

I assume that post combat damage event is fired after first strike damage because it should trigger twice if it's blocked by a first strike creature and a normal damage creature (or a double strike creature) but I'm not actually sure.  May be something you want to test.

Reply:
Tested with the current code and I'm currently getting separate triggers for first strike damage and normal combat damage.
Test the following scenarios.

First strike blocker + Normal blocker
Double strike blocker
Double strike blocker + First strike blocker + Normal blocker

All resulted with 2 separate triggers.

Reply:
You need to test two normal blockers though and make sure there's only one trigger.  I believe that's why the `COMBAT_DAMAGE_STEP_POST` event is used.

Reply:
Ah ok understood. Thanks for explaining, I redid the logic to mimic the `DealtDamageToSourceTriggeredAbility` logic and as far as I cans see it tests properly now.